### ISSUE TITLE
Ability to validate customization params at compile time

### ISSUE THREAD

**Issue by jmetzmeier** [hashicorp/otto#162]
Currently params such as ruby_version and not validated at compile time. If you enter something bogus for ruby_version (such as "foobar"), this will not bomb until the vagrant provisioning script tries to use that as the ruby version. It would be nice if some basic validation existed when "otto compile" is ran so that more friendly error messages can be presented.

**Comment by justincampbell** [comment]
Not sure how we could do this at compile-time, unless we simply check the overall format and not the specific version. For instance, if a new version is released (2.3), should we have to update Otto before anyone can use that version? Maybe we could just check `/\d\.\d/`.

**Comment by jmetzmeier** [comment]
@justincampbell Yes a simple check like /\d\.\d/ is what I was thinking. In pr #147 only a python major version can be used for python_version (eg 2.7 not 2.7.9). A simple regex check can catch when people try to enter a minor version. 

I am not sure that app/ruby is the correct label here. I believe that it should be core, as it is about enabling the possibility of these types of checks, rather than specifically adding a check to app/ruby.
